source/framework/vx_array.c

```diff
@@ -22,6 +22,182 @@ static void ownInitArrayObject(
+    return status;
+}
+
+/*! \brief This function is called to find out if it is OK to swap the input with the output.
+ * Both item type and capacity must be the same.
+ * \returns VX_SUCCESS if it is, otherwise another error code.
+ *
+ */
+static vx_status isArraySwapable(vx_array input, vx_array output)
+{
+    vx_status status = (vx_status)VX_SUCCESS;
+    tivx_obj_desc_array_t *ip_obj_desc = (tivx_obj_desc_array_t *)input->base.obj_desc;
+    tivx_obj_desc_array_t *op_obj_desc = (tivx_obj_desc_array_t *)output->base.obj_desc;
+    if (vx_false_e == ownIsValidInputAndOutputArrays(input, output))
+    {
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else if (op_obj_desc->item_type != ip_obj_desc->item_type)
+    {
+        /* Types must match */
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    else if (op_obj_desc->capacity != ip_obj_desc->capacity)
+    {
+        /* Output must have same capacity as input */
+        status = (vx_status)VX_ERROR_NOT_COMPATIBLE;
+    }
+    return status;
+}
+
+/*! \brief Copy input to output
+ * The input must be copyable to the output; checks done already.
+ * Note that locking a reference actually locks the context, so we only lock
+ * one reference!
+ */
+static vx_status copyArray(vx_array input, vx_array output)
+{
+    tivx_obj_desc_array_t *ip_obj_desc = (tivx_obj_desc_array_t *)input->base.obj_desc;
+    tivx_obj_desc_array_t *op_obj_desc = (tivx_obj_desc_array_t *)output->base.obj_desc;
+    vx_status status = (vx_status)VX_FAILURE;
+    if (ip_obj_desc->mem_size <= op_obj_desc->mem_size)
+    {
+        ownReferenceLock((vx_reference)output);
+        status = tivxMemBufferMap((void *)(uintptr_t)ip_obj_desc->mem_ptr.host_ptr, ip_obj_desc->mem_size, (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_READ_ONLY);
+        if (VX_SUCCESS == status)
+        {
+            status = tivxMemBufferMap((void *)(uintptr_t)op_obj_desc->mem_ptr.host_ptr, op_obj_desc->mem_size, (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_WRITE_ONLY);
+            if (VX_SUCCESS == status)
+            {
+                memcpy((void *)(uintptr_t)op_obj_desc->mem_ptr.host_ptr, (void *)(uintptr_t)ip_obj_desc->mem_ptr.host_ptr, ip_obj_desc->mem_size);
+                op_obj_desc->num_items = ip_obj_desc->num_items;
+                tivxMemBufferUnmap((void *)(uintptr_t)op_obj_desc->mem_ptr.host_ptr, op_obj_desc->mem_size, (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_WRITE_ONLY);
+
+            }
+            tivxMemBufferUnmap((void *)(uintptr_t)ip_obj_desc->mem_ptr.host_ptr, ip_obj_desc->mem_size, (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_READ_ONLY);
+        }
+        ownReferenceUnlock((vx_reference)output);
+    }
+    return status;
+}
+
+/*! \brief swap input and output pointers
+ * Input and output must be swappable; checks done already.
+ * Note that locking a reference actually locks the context, so we only lock
+ * one reference!
+ */
+static vx_status swapArray(vx_array input, vx_array output)
```

Comment:
GENERIC: You may also be able to generalize this for data types using vxSetMetaFormatFromReference.